archive_converter.py

```diff
@@ -13,8 +13,57 @@
 # limitations under the License.
 
 import email
+import os
 from setup_gmail import Message
 
+class ArchiveMessageIndex(object):
+    def __init__(self):
+        self._lookup_index = {} # Maps MessageKey to Thread
+        self._messages_seen = {} # Maps message.id to message
+
+    def find_thread(self, key: str) -> Message:
+        return self._lookup_index[key]
+
+    def update(self):
+        fully_updated = False
+        new_messages = []
```

Comment:
```suggestion
        new_messages : List[Message] = []
```

and similarly annotate `_add_messages_to_index()`